src/models/User.py

```diff
@@ -87,37 +84,38 @@ def poster(self, poster : str):
     def list_preferences(self):
         return self.__list_preferences
 
-    @list_preferences.setter
-    def list_preferences(self, list_preferences : List[Show]):
+    def __set_list_preferences(self, list_preferences : List[Show]):
         if type(list_preferences) is not List[Show] and list_preferences is not None:
             raise TypeError("List of preferences should be a list of preferences")
         else:
             self.__list_preferences = list_preferences
 
 
     def create_user_in_bdd(self, my_db: MyDBConnection):
-        my_db.exec_one("""
-        INSERT INTO user (surname, name, login, pwd, pict) VALUES 
-        ((?), (?), (?), (?), (?))""", (
-            self.firstname, self.surname, self.login, self.pwd, self.poster
-        ))
-        new_user = User.retrieve_user_from_credentials(self.login, self.pwd, my_db)
-        self.__setId(new_user.id)
+        new_user=self.retrieve_user_from_credentials(self.login,self.pwd,my_db)
+        if new_user==None:
+            my_db.exec_one("""
+            INSERT INTO user (surname, name, login, pwd, pict) VALUES 
+            ((?), (?), (?), (?), (?))""", (
+                self.firstname, self.surname, self.login, self.pwd, self.poster
+            ))
+            new_user = User.retrieve_user_from_credentials(self.login, self.pwd, my_db)
+            self.__set_id(new_user.id)
 
     def update_user_in_bdd(self, my_db: MyDBConnection, firstname: str = None, surname: str = None, login: str = None, pwd: str = None,
                            poster: str = None, list_preferences: List[Show]= None):
         if firstname is not None:
-            self.firstname = firstname
+            self.__set_firstname(firstname)
         if surname is not None:
-            self.surname = surname
+            self.__set_surname(surname)
         if login is not None:
-            self.__setLogin(login)
+            self.__set_login(login)
         if pwd is not None:
-            self.__setPwd(pwd)
+            self.__set_pwd(pwd)
         if poster is not None:
-            self.poster = poster
+            self.__set_poster(poster)
         if list_preferences is not None:
-            self.list_preferences = list_preferences
+            self.__set_list_preferences(list_preferences)
         my_db.exec_one("UPDATE USER SET firstname=(?), surname=(?), login=(?), pwd=(?), poster=(?) WHERE id=(?)", 
```

Comment:
The call to the DB is not correct...
```suggestion
        my_db.exec_one("UPDATE USER SET surname=(?), name=(?), login=(?), pwd=(?), pict=(?) WHERE id=(?)", 
```

Reply:
Actually modified in the DB so OK now ^^